pkgs/stdenv/generic/builder.sh

```diff
@@ -1,10 +1,12 @@
+set -o errexit -o noclobber -o nounset
```

Comment:
I don't think we can enable noclobber. If I understand the bash man correct then this would make every valid use case where you are overriding a file with a redirect fail.

---

Question: Why are we no using the short options? At least in my impression ``set -eou pipefail`` is pretty well known compared to ``set -o errexit -o pipefail -o nounset``.

Reply:
Yes, repeated `> foo`s would fail. I'd consider that a code smell (why reuse the same file during a single build?), but I guess we'll see if that happens.

IMO `set -o errexit -o pipefail -o nounset` is readable, while `set -eou pipefail` is not. Also, `set -eou pipefail` is not actually what you want, it's `set -euo pipefail`, so it's easier to get wrong :wink: .

Reply:
> Yes, repeated > foos would fail.

You mean by repeated when I redirect twice to foo or also when I redirect to foo and that file was already in the source tarball?

> why reuse the same file during a single build?

Sure it is not ideal but bug hunting packages that do that and then solve the issue is also not ideal. I think we have way bigger code smells in the shell scripts which cause way more bugs. One that come in mind is that quoting of the most args like options in mkDerivation does not matter to much.

> I guess we'll see if that happens.

The main problem here is that we would need to rebuild (almost) every package to make sure that nothing major breaks and then researching if it was this change or another one on staging is probably not fund and takes a while.

> Also, set -eou pipefail is not actually what you want, it's set -euo pipefail, so it's easier to get wrong 😉 .

I had that discussion before. It does not matter. Both works.

You can try that for example like so:
```ShellSession
$ (set -oeu pipefai)
bash: set: pipefai: invalid option name
```

Reply:
> > Yes, repeated > foos would fail.
>
> You mean by repeated when I redirect twice to foo or also when I redirect to foo and that file was already in the source tarball?

I don't understand what you mean by this. What is that reference to the source tarball? What I'm saying is simply that

```
> foo
[some other code]
> foo
```

will fail.

I think you meant `(set -oeu pipefail)`. Anyway, you're right, `diff <(set -oeu pipefail; set) <(set -ueo pipefail; set)` is empty. TIL.

Reply:
> I don't understand what you mean by this. What is that reference to the source tarball?

``touch foo`` and then echo > foo`` does not fail?

Reply:
Yes, that'll fail too.

Reply:
I've removed `-o noclobber` for now. We could always look into it later.